projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -235,11 +239,13 @@ export class MultiSelectComponent<V> implements ControlValueAccessor, AfterConte
 
   public ngAfterContentInit(): void {
     this.allOptions$ = this.allOptionsList !== undefined ? queryListAndChanges$(this.allOptionsList) : EMPTY;
-    this.filteredOptions$ = combineLatest([this.allOptions$, this.caseInsensitiveSearchSubject]).pipe(
-      map(([options, searchText]) =>
-        isEmpty(searchText)
-          ? options.toArray()
-          : options.filter(option => option.label.toLowerCase().includes(searchText.toLowerCase()))
+    this.filteredOptions$ = combineLatest([
+      this.allOptions$,
+      this.caseInsensitiveSearchSubject,
+      this.reorderSelectedItemsSubject
+    ]).pipe(
+      map(([options, searchText, reorderSelectedItemsSubject]) =>
+        this.getFilteredOptions(options, searchText, reorderSelectedItemsSubject)
```

Comment:
we don't need to do anything special here. we just have to put the selected items on top of the list always.

Reply:
done